Added HSV Adjust Dialog

luiq54 opened the pull request:
It works for selection too!
Tests Needed!

OverloadedOrama commented:
This closes #188 right?

Also, could you rebase your branch so it won't be behind Orama-Interactive:master? There are conflicts in Main.tscn

luiq54 commented:
@OverloadedOrama I have rebased it and run some tests to make sure it works! 😄

OverloadedOrama commented:
It seems to be a bit buggy if I press Cancel, and then redo. The idea of undoing on Cancel is a bit problematic. I think the best way to do it would be with a preview panel, like rotation has. If cancel is pressed, nothing happens. If OK is pressed, the changes happen. I plan to put previews on all other Image effects, probably for v0.8.

I've also noticed problems with the sliders. If I draw something black and then make Value go to 99, the resulting color is gray. If I drag the Value all the way to 0 and then back to 99, the resulting color is white. Shouldn't the same values result in the same results?

luiq54 commented:
Yep, Preview panel seems the better option. This wil actually solve the second problem as well. Will add those today or tomorrow!

luiq54 commented:
@OverloadedOrama I have added the previews and fixed the bugs as well. Also, the undo/redo functionality has been modified. Check it out and tell me if you encounter any more bugs! 😄

P.S. Dont merge it yet.  I am planning to add another commit regarding changes mentioned in contributing guidelines.

OverloadedOrama commented on Scripts/Canvas.gd:
```diff
@@ -1010,3 +1010,85 @@ func blend_rect(bg : Image, brush : Image, src_rect : Rect2, dst : Vector2) -> v
+						elif delta < 0:
+							sat = range_lerp(delta,-100,0,0,c.s)
+						c.s = sat
+						print(sat)
```
This is causing a lot of lag, make sure it remove it before the PR gets merged. Printing values multiple times in a frame generally causes a lot of lag, so make sure to only use it for specific debugging purposes and remove them before commiting! 😄

luiq54 replied:
oops,  my bad! 😅 . Yep I will fix it !!

OverloadedOrama commented on Scripts/Canvas.gd:
```diff
@@ -1010,3 +1010,85 @@ func blend_rect(bg : Image, brush : Image, src_rect : Rect2, dst : Vector2) -> v
+						elif delta < 0:
+							val = range_lerp(delta,-100,0,0,c.v)
+						
+						c.s = val
```
Small typo here, this should be `c.v`, not `c.s`.

luiq54 commented:
Fixed the bugs and went through the code once! I have added Translations Strings. I think this is ready to merge.

OverloadedOrama commented:
Yeah I think so too. Thank you!
